modular/gater/admin_handler.go

```diff
@@ -877,27 +877,42 @@ func (g *GateModular) getRecoverPiece(ctx context.Context, objectInfo *storagety
 		sspAddress = append(sspAddress, sp.OperatorAddress)
 	}
 
-	isSuccessor := false
 	var (
-		successorSP *sptypes.StorageProvider
-		swapInInfo  *virtualgrouptypes.SwapInInfo
+		isSuccessorPrimary   bool
+		isSuccessorSecondary bool
+		successorSP          *sptypes.StorageProvider
+		swapInInfo           *virtualgrouptypes.SwapInInfo
 	)
 	if recoveryTask.GetBySuccessorSp() {
-		swapInInfo, err = g.baseApp.Consensus().QuerySwapInInfo(ctx, gvg.FamilyId, 0)
-		if err != nil {
-			return nil, ErrConsensusWithDetail("query swapInInfo err: " + err.Error())
-		}
-		successorSP, err = g.baseApp.Consensus().QuerySPByID(ctx, swapInInfo.SuccessorSpId)
-		if err != nil {
-			return nil, ErrConsensusWithDetail("query sp err: " + err.Error())
-		}
-		if primarySp.Id == swapInInfo.TargetSpId && successorSP.OperatorAddress == signatureAddr.String() {
-			isSuccessor = true
+		swapInInfo, err = g.baseApp.Consensus().QuerySwapInInfo(ctx, gvg.FamilyId, virtualgrouptypes.NoSpecifiedGVGId)
```

Comment:
Need we update test file to cover and verify this long section of changes?

Reply:
make it compatible for more case, so the test wont break.